src/components/Hidden/Hidden.js

```diff
@@ -61,6 +68,14 @@ class Component extends React.Component {
     }
   };
 
+  handleClickOutside = e => {
+    const { className, hide, visible } = this.props;
+
+    if (!e.target.closest(`[class^='${className}']`) && visible && hide) {
```

Comment:
I think this won't work if the clicked element has the same class as this one. 

Here is an alternative approach that uses `Node.contains` to perform the check which is slightly more bullet proof: https://codesandbox.io/s/94kwvn31oy

It still has the drawback though that it won't work through [portals](https://reactjs.org/docs/portals.html). If a child component of the <Hidden /> component renders something to a portal, (such as a modal or dropdown or something) then clicks inside the Portal tree will not be considered to be "inside" the <hidden /> components DOM hierarchy and thus will be considered "outside" clicks. I have implemented workarounds to this in a different project but I'm not sure how they should be implemented in this library so that they are easy to use and unsurprising.

Reply:
> Here is an alternative approach that uses Node.contains to perform the check which is slightly more bullet proof: https://codesandbox.io/s/94kwvn31oy

Interesting, I didn't realize that will not work in this case, your solution probably is the best

> I have implemented workarounds to this in a different project but I'm not sure how they should be implemented in this library so that they are easy to use and unsurprising.

Can you please share this workarounds with us for discussion and tests?

Reply:
Here is the implementation of what we do at bugsnag: 

https://gist.github.com/wordofchristian/78348722edd275880dde54c549a0f155

We call it ClickBoundary. Every ClickBoundary uses context to keep track of any other click boundaries that are descendants of it (context will work across portals). The key is that it only works if the content of the portal is also wrapped in a ClickBoundary, if not, it won't get added to the tree of tracked nodes. So it's not entirely bullet proof from a usage standpoint.